src/System.Private.CoreLib/shared/System/Threading/CancellationToken.cs

```diff
@@ -189,7 +193,7 @@ public CancellationToken(bool canceled) : this(canceled ? CancellationTokenSourc
         /// <returns>The <see cref="T:System.Threading.CancellationTokenRegistration"/> instance that can 
         /// be used to unregister the callback.</returns>
         /// <exception cref="T:System.ArgumentNullException"><paramref name="callback"/> is null.</exception>
-        public CancellationTokenRegistration Register(Action<object> callback, object state) =>
+        public CancellationTokenRegistration Register(Action<object?> callback, object? state) =>
```

Comment:
@jcouv, there are a bunch of places in the framework where we take an `Action<object>` and an `object` state to pass to it.  Thus, the `object` in `Action<object> callback` will be null if and only if `state` is null.  Do we have any attribution planned that will allow for that relationship to be expressed?  I don't et know if it would actually matter much, so just asking right now.

Reply:
We have not discussed any such attribute yet.
Feel free to add a comment with scenario details in https://github.com/dotnet/roslyn/issues/26761